spec/client_spec.rb

```diff
@@ -408,53 +408,82 @@
+    let(:default_headers) { { 'Content-Type' => 'application/json' } }
 
-    let(:perform) { subject.monitor_topics }
-    let(:expected_result) do
-      [
-        {
-          name: 'widgets',
-          publisher: 'demo',
-          events: 12589
-        }
-      ]
+    shared_context 'successful connection to bus' do
+      before do
+        @stub = stub_request(:get, url)
+          .with(basic_auth: [options[:uuid], 'x'], headers: default_headers)
+          .to_return(status: 200, body: expected_result.to_json)
+      end
     end
 
-    context 'the connection to the bus is successful' do
+    shared_context 'failing connection to bus' do
       before do
-        @stub = stub_request(:get, 'https://bus.example.com/topics').
-          with(basic_auth: [options[:uuid], 'x']).
-          with { |r|
-          r.headers['Content-Type'] == 'application/json'
-        }.to_return {
-          { status: 200, body: expected_result.to_json }
-        }
+        @stub = stub_request(:get, url)
+          .with(basic_auth: [options[:uuid], 'x'], headers: default_headers)
+          .to_return(status: 500)
       end
+    end
 
-      it 'expects a collection of topics' do
-        expect(perform.map(&:attributes)).to eql(expected_result)
+    describe '#monitor_topics' do
+      let(:url) { 'https://bus.example.com/topics' }
+      let(:perform) { subject.monitor_topics }
+      let(:expected_result) do
+        [
+          {
+            name: 'widgets',
+            publisher: 'demo',
+            events: 12589
+          }
+        ]
       end
-    end
 
-    context 'the connection to the bus errors' do
-      before do
-        @stub = stub_request(:get, 'https://bus.example.com/topics').
-          with(basic_auth: [options[:uuid], 'x']).
-          with { |r|
-          r.headers['Content-Type'] == 'application/json'
-        }.to_return(status: 500)
+      context 'the connection to the bus is successful' do
+        include_context 'successful connection to bus'
+
+        it 'expects a collection of topics' do
+          expect(perform.map(&:attributes)).to eql(expected_result)
+        end
       end
 
-      it 'expects a collection of topics' do
-        expect { perform }.to raise_error(RuntimeError)
+      context 'the connection to the bus errors' do
+        include_context 'failing connection to bus'
+
+        it 'expects a collection of topics' do
+          expect { perform }.to raise_error(RuntimeError)
+        end
       end
     end
-  end
 
-  describe '#monitor_subscriptions' do
-    it 'passes'
-  end
+    describe '#monitor_subscriptions' do
+      let(:url) { 'https://bus.example.com/subscriptions' }
+      let(:perform) { subject.monitor_subscriptions }
+      let(:expected_result) do
+        [{
+          subscriber: 'bob',
+          callback:   'https://app.example.com/events',
+          topics:     ['widgets', 'kitten'],
+          events:     { sent: 1, queued: 100, oldest: 10_000 }
```

Comment:
Definitely worthwhile, especially since we do have the data for it !
If want to do it add a ticket in the [current RM epic](https://deliveroo.atlassian.net/browse/SETI-65)?